### ISSUE TITLE
make base64(password) and other options

### ISSUE THREAD

**Issue by lasers** [ultrabug/py3status#1211]
I'm not able to use `password:base64` with `mail` https://github.com/ultrabug/py3status/pull/1210 (only for `imap`) because `accounts` is a dict consisting of mailbox types and a list of dicts consisting of mailbox settings and/or paths to use. I think we need to make `base64(password)` similar to `env()` and any other decoders or options in the future should follow this style too.
```python
mail {
    accounts = {
        'imap': [
            {
                'user': 'lasers',
                'pass': base64('a2lzc19teV9idXR0IQo='),
                'server': 'imap.gmail.com',
                'port': 993,
            },
        ]
    }
}
```
EDIT: Hi. I got a working `base64 decoder`. Not perfect as I don't know what to use in `re.match` and I can trip on some things such as `order += base64(c3RhdGljX3N0cmluZwo=)` too. I got everything from `env()`. https://github.com/lasers/py3status/tree/add-base64-decoder.

**Comment by tobes** [comment]
This is an interesting idea.  Really it'd be nice to have a good solution to private stuff maybe using a keyring or similar.  I'll have a think on this.

Your regex looks wrong to me, as some valid chars look missing `+/` as well as trailing `=`.  It could also be much more compact (readable)

**Comment by lasers** [comment]
I'd like to resolve the following issues first before jumping on `keyring` or similar, eg `pass`. 
1) Resolve `mail` #1210 issue right away.
a) This would close the second last issue on the list.
b) This would close the last pull request on the list.
2) Resolve `base64` issue (this)... mainly for `imap` users with a security concern.
3) Any other pull requests.
a) Resolve the easy ones, eg typos, annihilation, etc.
b) New modules. What's the deal here? :-o
4) See where we are at this point... then revisit some things.
a) What to do with the new formats?
b) Your `keyring` / `pass` solution. 

I have a branch + enhancement to close the last issue on the list: `pomodoro format`.

**Comment by tobes** [comment]
> . The new PR broke things.

Please give details

**Comment by lasers** [comment]
I got `Bad type conversion` notification. I can try and look into this later.

**Comment by tobes** [comment]
check your base64 is valid just in case.  I'm assuming it is the `base64()` you tried?  did you specify a conversion?

**Comment by lasers** [comment]
Closed via #1498 with a nice cherry #1503 on top.
